### ISSUE TITLE
python build No matching distribution found for isoduration; extra == "format"

### ISSUE THREAD

**Issue by sgiroux-spwr** [NixOS/nixpkgs#176931]
### Steps To Reproduce
1. `nixops deploy --check --repair -I nixpkgs=channel:nixos-22.05 --include`

### Build log
```
...
Requirement already satisfied: jsonref in /nix/store/jgy18iwhiqpfb3imc9r8gwnlhs7c7v3d-python3.9-jsonref-0.2/lib/python3.9/site-packages (from bravado-core==5.17.0) (0.2)
Requirement already satisfied: swagger-spec-validator>=2.0.1 in /nix/store/0aapc4qmfgpb6sls3h00d91x6qqi2a2r-python3.9-swagger-spec-validator-2.7.4/lib/python3.9/site-packages (from bravado-core==5.17.0) (2.7.4)
Requirement already satisfied: jsonschema[format]>=2.5.1 in /nix/store/08g1n9bp2lk4m6ag9kcslylcmvd1wrdr-python3.9-jsonschema-4.5.1/lib/python3.9/site-packages (from bravado-core==5.17.0) (4.5.1)
Requirement already satisfied: msgpack>=0.5.2 in /nix/store/10n6r33zv0apbwfncz5hjxkfvsb9c5z9-python3.9-msgpack-1.0.3/lib/python3.9/site-packages (from bravado-core==5.17.0) (1.0.3)
Requirement already satisfied: simplejson in /nix/store/z82yj60650rf4kf1bdnlbpyza8pv8xhy-python3.9-simplejson-3.17.6/lib/python3.9/site-packages (from bravado-core==5.17.0) (3.17.6)
Requirement already satisfied: six in /nix/store/n4pljljbihyvw2psc6k55dgpnbwdg4lz-python3.9-six-1.16.0/lib/python3.9/site-packages (from bravado-core==5.17.0) (1.16.0)
Requirement already satisfied: pytz in /nix/store/43dsisfzbm57czlxpb1nfcvwym2j842n-python3.9-pytz-2022.1/lib/python3.9/site-packages (from bravado-core==5.17.0) (2022.1)
Requirement already satisfied: pyyaml in /nix/store/4hb8jdyimfjkp22qjfdjg4s2q5bmsfb4-python3.9-pyyaml-6.0/lib/python3.9/site-packages (from bravado-core==5.17.0) (6.0)
Requirement already satisfied: python-dateutil in /nix/store/xwymfylq4lkxvzy1f6s4cr7frc62df2s-python3.9-python-dateutil-2.8.2/lib/python3.9/site-packages (from bravado-core==5.17.0) (2.8.2)
Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in /nix/store/w3yc59w85alx30sv14p3lx4441bxyvhw-python3.9-pyrsistent-0.18.1/lib/python3.9/site-packages (from jsonschema[format]>=2.5.1->bravado-core==5.17.0) (0.18.1)
Requirement already satisfied: attrs>=17.4.0 in /nix/store/dd9c95s9jgg6m6hgscn1am9n11d3ws1f-python3.9-attrs-21.4.0/lib/python3.9/site-packages (from jsonschema[format]>=2.5.1->bravado-core==5.17.0) (21.4.0)
INFO: pip is looking at multiple versions of <Python from Requires-Python> to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of bravado-core to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement isoduration; extra == "format" (from jsonschema[format]) (from versions: none)
ERROR: No matching distribution found for isoduration; extra == "format"
...
```

### Additional context
Effectively, I'm building a python package that requires `bravado`, which requires a number of sub-dependencies until `jsonschema[format]` but I believe something is applying the 'extra' [format] to other dependencies (like isoduration)

### Metadata
```console
# nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.94, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(root): `"nixos-21.11.335733.f6ddd55d5f9"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
 ```

### The offending private python package:
pyproject.toml
```
[tool.poetry]
name = "redacted"
version = "0.1.1"
description = "redacted"
authors = ["redacted <redacted>"]

[tool.poetry.dependencies]
python = "^3.9"
six = "^1.15.0"
bravado = "^11.0.3"
pyzmq = "^22.0.3"
docker = "^5.0.3"
paramiko = "^2.10.5"
Jinja2 = "^3.0.2"
netifaces = "^0.11.0"
scp = "^0.14.1"
pydantic = "^1.8.2"
```

**Comment by fabaff** [comment]
`bravado-core` was fixed on `master` in #174952 but it looks like the backport to 22.05 [failed](https://github.com/NixOS/nixpkgs/pull/174952#issuecomment-1148467402).

**Comment by sgiroux-spwr** [comment]
> `bravado-core` was fixed on `master` in #174952 but it looks like the backport to 22.05 [failed](https://github.com/NixOS/nixpkgs/pull/174952#issuecomment-1148467402).

Sounds good to me -- I had just found that PR. I'll see what I can do

**Comment by sgiroux-spwr** [comment]
See #177017
